galsim/chromatic.py

```diff
@@ -1226,6 +1227,90 @@ def __init__(self, original, waves, oversample_fac=1.0, use_exact_sed=True,
 
         # Don't interpolate an interpolation.  Go back to the original.
         self.deinterpolated = original.deinterpolated
+    
+    @classmethod
+    def from_images(self, images, waves, _force_stepk = None, _force_maxk = None, **kwargs):
+        """
+        Alternative initiazliation to InterpolatedChromaticObject from input images at specific wavelenghts. Any parameters accepted by
+        the InterpolatedImage class can be passed in kwargs. Note that stepk and maxk are parameters that can depend on the image size,
+        and therefore on the wavelength. If not given, as a list for every image, or a single number for all images, it will be caluclated
+        using the input image pixel scale and dimensions. This means it will be identical for all images. This will cause small differences
+        from the normal use of this class using chromatic objects whose stepk and maxk are wavelength-dependent. To avoid sanity checks
+        from method initialization, such as wavelength sorting, pixel scale and image dimensions compatibility, simply call the function
+        InterpolatedChromaticObject._from_images directly.
+        
+        Parameters:
+            images:          list of Galsim Image objects to use as interpolated images. All images must have the same pixel scale and image
+                             dimensions.
+            wave:            list of wavelength values in nanometers.
+            _force_stepk:    list of step_k values to pass to InterpolatedImages for each image. Can also be single value
+                             to pass for all images alike. If not given stepk is calculated from image pixel scale and dimensions. [Default: None]
+            _force_maxk:     list of max_k values to pass to InterpolatedImages for each image. Can also be single value
+                             to pass for all images alike. If not given maxk is calculated from image pixel scale. [Default: None]
+
+        Returns:
+            An InterpolatedChromaticObject intialized from input images.
+        """ 
+        # check that all images have the same pixel scale and image dimensions. 
+        pix_scale = images[0].scale
```

Comment:
If someone uses images with a non-PixelScale wcs, this will give a possibly obtuse error message.  Maybe worth adding:
```
if images[0].wcs is None or not images[0].wcs._isPixelScale:
    raise GalSimValueError("images must have a pixel scale wcs.")
```

Reply:
Added.